frontend_server/server.js

```diff
@@ -32,6 +34,29 @@ const HOST = '0.0.0.0';
 
 const app = express();
 
+// Add Authorization header for all requests that are proxied to the data server.
+// TODO: The token can be cached and only refreshed when needed
```

Comment:
Do you know how much of a performance hit getting the token is? These requests are performance-sensitive since we depend on them to show visualizations

Reply:
Cannot find documentation on added latency. I will file an issue for caching the token to improve performance.